src/cli/linker.zig

```diff
@@ -113,11 +139,32 @@ pub fn link(allocator: Allocator, config: LinkConfig) LinkError!void {
             try args.append("-o");
             try args.append(config.output_path);
 
-            // Suppress LLD warnings
+            // Prevent hidden linker behaviour -- only explicit platfor mdependencies
+            try args.append("-nostdlib");
+            // Remove unused sections to reduce binary size
+            try args.append("--gc-sections");
+            // TODO make the confirugable instead of using comments
+            // Suppress linker warnings
             try args.append("-w");
-
-            // Use static linking to avoid dynamic linker dependency issues
-            try args.append("-static");
+            // Verbose linker for debugging (uncomment as needed)
+            // try args.append("--verbose");
+            // try args.append("--print-map");
+            // try args.append("--error-limit=0");
+
+            // Determine target ABI
+            const target_abi = config.target_abi orelse if (builtin.target.abi == .musl) TargetAbi.musl else TargetAbi.gnu;
+
+            switch (target_abi) {
+                .musl => {
+                    // Static musl linking
+                    try args.append("-static");
+                },
+                .gnu => {
+                    // Dynamic GNU linking
+                    try args.append("-dynamic-linker");
+                    try args.append("/lib64/ld-linux-x86-64.so.2");
+                },
+            }
```

Comment:
This is good enough for proof of concept, but needs improvement. I'm not sure what yet.